internal/service/oauth_protected_resource_validation_test.go

```
@@ -0,0 +1,170 @@
+func TestBoundedOAuthHTTPClientConstrainsTimeoutsAndRedirects(t *testing.T) {
+	previousCalled := false
+	client := boundedOAuthHTTPClient(&http.Client{
+		Timeout: time.Hour,
+		CheckRedirect: func(*http.Request, []*http.Request) error {
+			previousCalled = true
+			return errors.New("caller rejected redirect")
+		},
+	})
+	require.Equal(t, oauthProviderTimeout, client.Timeout)
+
+	request, err := http.NewRequest(http.MethodGet, "https://issuer.example/jwks", nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)